src/NHibernate/Proxy/NHibernateProxyBuilder.cs

```diff
@@ -393,9 +391,28 @@ private static void ImplementGetLazyInitializer(TypeBuilder typeBuilder, MethodI
 			IL.Emit(OpCodes.Ldnull);
 			IL.Emit(OpCodes.Bne_Un, skipBaseCall);
 
-			IL.Emit(OpCodes.Ldarg_0);
-			EmitCallMethod(IL, OpCodes.Call, method);
-			IL.Emit(OpCodes.Ret);
+			if (method.IsAbstract)
+			{
```

Comment:
Thinking about it again, this could be an unlikely breaking change. That requires some weird class, but still.
Consider:
```c#
public interface IA
{
    int Id { get; set; }
}

public class A: IA
{
    public virtual int Id { get; set; }

    public A()
    {
        IA ia = this;
        ia.Id = -1;
        Assert(ia.Id, Is.EqualTo(-1);
    }
}
```

With previous logic, it would work, since the interface call was redirected to the implicit implementation.

With the new logic, it will fail, since the interface call will now be no-op and yielding default value, even when an implicit implementation is available.

I think it is such a corner case that it is acceptable. Moreover supporting this case robustly would require a likely costly search for a matching implicit implementation on the base type, in case the interface is explicitly implemented by it. (I know that other case is not yet supported by the way, due to the way interface method implementations are currently declared.)

Still I add a possible breaking change here.

Reply:
v5.1 one. About possible breaking changes, only released versions count.